build/common.ps1

```diff
@@ -257,28 +253,43 @@ Function Test-MSBuildVersionPresent {
     Test-Path $MSBuildExe
 }
 
-$MSBuildExe = Get-MSBuildExe
-Set-Alias msbuild $MSBuildExe
-
-$VS14Installed = Test-MSBuildVersionPresent -MSBuildVersion 14
-$VS15Installed = Test-MSBuildVersionPresent -MSBuildVersion 15
-
-function Test-BuildEnvironment {
+Function Test-BuildEnvironment {
     param(
         [switch]$CI
     )
+    if (-not (Test-Path $ConfigureJson)) {
+        Error-Log 'Build environment is not configured. Please run configure.ps1 first.' -Fatal
+    }
+
     $Installed = (Test-Path $DotNetExe) -and (Test-Path $NuGetExe)
     if (-not $Installed) {
         Error-Log 'Build environment is not configured. Please run configure.ps1 first.' -Fatal
     }
 
+    $script:ConfigureObject = Get-Content $ConfigureJson -Raw | ConvertFrom-Json
+    Set-Variable MSBuildExe -Value $ConfigureObject.BuildTools.MSBuildExe -Scope Script -Force
+    Set-Alias msbuild $script:MSBuildExe -Scope Script -Force
+    Set-Variable BuildToolsets -Value $ConfigureObject.Toolsets -Scope Script -Force
+
+    $script:VS14Installed = ($BuildToolsets | where vs14 -ne $null)
+    $script:VS15Installed = ($BuildToolsets | where vs15 -ne $null)
+
+    $ConfigureObject |
+         select -expand envvars |
+         %{ $_.psobject.properties } |
+         %{ Set-Item -Path "env:$($_.Name)" -Value $_.Value }
+
     if ($CI) {
         # Explicitly add cli to environment PATH
         # because dotnet-install script runs in configure.ps1 in previous build step
         $env:path = "$CLIRoot;${env:path}"
     }
 }
 
+Function Use-BuildToolset {
```

Comment:
A placeholder for a method to switch toolsets between build steps, e.g for setting environment variables. Will remove for now.